ControlRoomApplication/ControlRoomApplication/Controllers/ControlRoomController/RadioTelescopeControllerManagementThread.cs

```diff
@@ -163,21 +173,28 @@ public void InterruptOnce()
         private void SpinRoutine()
         {
             bool KeepAlive = KeepThreadAlive;
-            
+            //Appointment OldAppointment = new Appointment();
+
+
             while (KeepAlive)
             {
-                logger.Info("Waiting for next Appointment");
-                Appointment NextAppointment = WaitForNextAppointment();
+                //Appointment NextAppointment = WaitForNextAppointment();
+                NextAppointment = WaitForNextAppointment();
+                if (NextAppointment != OldAppointment)
```

Comment:
Because `NextAppointment` and `OldAppointment` are objects, the `!=` operator will only tell us that their memory addresses are different. Technically, this still yields the same result, but we want to try to equate them in a way we can see in C# to prevent possible bugs down the road. To fix it, I suggest comparing each `Appointment`'s `Id` field, since the ID is unique for each appointment.